plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineDeployOutput.java

```diff
@@ -27,7 +29,13 @@
 // TODO: move into appengine-plugins-core
 // TODO expand to include other Version attributes
 public class AppEngineDeployOutput {
-  private AppEngineDeployOutput() {
+  @VisibleForTesting
+  AppEngineDeployOutput(String id, String service) {
```

Comment:
You don't have to make this @VisibleForTesting. Instead of calling the constructor directly, you could pass JSON into parse()